### PR TITLE
Add Stat Component

### LINKED ISSUE #186: Stat
The `Stat` component is used to style inline text representing data or statistics. In "shops.seas", this would be used in the historical attendance table to represent the number of sign-ins for a given day.

This should be rendered in a monospace font.

### PR DESCRIPTION
**Pull request by amalyam**
## Describe your changes
This PR adds a `Stat` component which styles inline text representing data or statistics in a monospace font.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:
- [x] I have run `eslint` on the code
- [x] I have added JSDoc for all of my code (where applicable)
- [ ] I have added tests to cover my changes.

## Priority:
- [x] Normal
- [ ] High

## Related Issues:
Closes #186

### CODE REVIEW

**Review by jonseitz — CHANGES_REQUESTED** [review]
Nice work! I have Just a couple things on the component, and then a suggestion for adding a little more context to the documentation with a real usage example.

File: src/Typography/Stat.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+interface StatProps {
+  /**
+   * The data or statistics to be styled by this component
+   */
+  children: ReactNode;
+}
+
+const StyledStat = styled.div<StatProps>`
+  font-family: ${({ theme }): string => theme.font.data.family};
```
[the remark this answers is not in the document]
Comment (amalyam) [reply src/Typography/Stat.tsx:12]: Got it, I wasn't sure if we wanted more user control over that or not!

File: examples/Typography/Stat.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Can be used to style inline text representing data or statistics.
+
+ ```jsx
+ <Stat>100%</Stat>
+ ```
```
[the remark this answers is not in the document]
Comment (amalyam) [reply examples/Typography/Stat.md:5]: Thanks for laying this all out! Happy to take care of something like this next time!

File: src/Typography/Stat.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonseitz) [reply src/Typography/Stat.tsx:12]: Yeah, that makes sense. Maybe not the exact same thing, but Styled Components does have a way to [extend styles][exsty] by wrapping existing components in `styled()` and applying new CSS props.


[exsty]: https://styled-components.com/docs/basics#extending-styles

**Review by jonseitz — APPROVED** [review]
This is all set! :rocket:

File: src/Typography/Stat.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (amalyam) [reply src/Typography/Stat.tsx:12]: Ah ok, good to know!

**Pull request merged**